### DISCUSSION TITLE
Client cannot boot after download NBP file

### DISCUSSION THREAD

**Discussion by smallfish01** [ipxe/ipxe#500] [General]
I met the error when I client boot from iPXE, the client can download the NBP file(ipxe.efi), but after downloaded the file and the client no next step just go back to BIOS boot menu.

The iPXE server tftp log as below:

# tcpdump -n -i eth0 port 69
dropped privs to tcpdump
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes

17:39:37.106043 IP 192.168.19.164.informatik-lm > 192.168.19.250.tftp:  30 RRQ "ipxe.efi" octet blksize 1468
17:39:43.299809 IP 192.168.19.164.nms > 192.168.19.250.tftp:  30 RRQ "ipxe.efi" octet blksize 1468
17:39:47.003500 IP 192.168.19.164.tpdu > 192.168.19.250.tftp:  30 RRQ "ipxe.efi" octet blksize 1468
17:39:49.799864 IP 192.168.19.164.rgtp > 192.168.19.250.tftp:  30 RRQ "ipxe.efi" octet blksize 1468

# tail -f /var/log/messages

Oct 19 17:20:52 raspberrypi-4B dhcpd[3839]: DHCPDISCOVER from 70:b5:e8:35:62:75 via eth0
Oct 19 17:20:53 raspberrypi-4B dhcpd[3839]: DHCPOFFER on 192.168.19.173 to 770:b5:e8:35:62:75 via eth0
Oct 19 17:20:55 raspberrypi-4B dhcpd[3839]: DHCPREQUEST for 192.168.19.173 (192.168.19.250) from 70:b5:e8:35:62:75 via eth0
Oct 19 17:20:55 raspberrypi-4B dhcpd[3839]: DHCPACK on 192.168.19.173 to 70:b5:e8:35:62:75via eth0
Oct 19 17:20:55 raspberrypi-4B systemd[1]: Started Tftp Server.
Oct 19 17:20:55 raspberrypi-4B in.tftpd[43779]: RRQ from ::ffff:192.168.19.173 filename ipxe.efi
Oct 19 17:20:55 raspberrypi-4B in.tftpd[43779]: tftp: client does not accept options
Oct 19 17:20:55 raspberrypi-4B in.tftpd[43780]: RRQ from ::ffff:192.168.19.173 filename ipxe.efi
Oct 19 17:20:55 raspberrypi-4B in.tftpd[43780]: Client ::ffff:192.168.19.173 finished ipxe.efi

I have rebuild ipxe.efi file and cannot resolve the issue, no any idea.

The client reboot screen as the attachment.
![20211020190747](https://user-images.githubusercontent.com/4077255/138082197-76330402-ca7d-4367-bef9-89d36584ea2e.jpg)
![20211020190738](https://user-images.githubusercontent.com/4077255/138082217-3d852607-c723-47e5-8a34-8876a0a65b84.jpg)

Thank you.

**Comment by mcb30** [comment]
Most likely you either have Secure Boot enabled, or are using an `ipxe.efi` built for the wrong CPU architecture (e.g. using an x86_64 build on an arm64 machine)

**Reply by smallfish01** [reply]
Thanks for you replied. I will check my client PC if enable secure boot, and I'd like to know my client PC is Intel CPU and my iPXE server is Raspberry Pi which installed Centos 8, I tried build ipxe.efi from another Intel CPU architecture Centos server and ARM64 but both ipxe.efi files didn't work as well.
So what CPU architecture ipxe.efi file should I build ?

Thank you very much.

**Reply by mcb30** [reply]
To boot a client PC with an Intel CPU, you will need the normal x86_64 build of `ipxe.efi`.  You can download a ready-made version from https://boot.ipxe.org/ipxe.efi

**Reply by smallfish01** [reply]
I checked my client PC BIOS setting, it enabled secure boot, just disabled and reboot, it works now! thank you very much!

The ipxe.efi which your provided works as well.

Thank you again, I really appreciate your great help! Sorry for the trouble.

Have a good day.

Thank you.
